### PR TITLE
Contextual Sign In: Added homepage example

### PR DESCRIPTION
**Pull request by Garneauma**
Added example for contextual sign in button on the homepage (adjusted homepage layout to be consistent with other pages when authentication is enabled).

### CODE REVIEW

File: sites/authentication/_base.scss [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 	.btn {
 		border-radius: 0px;
-		margin-top: 5px;
+		margin-top: 3px;
```
Comment (GormFrank) [inline sites/authentication/_base.scss:9]: @Garneauma Je crois que ça vaudrait la peine de séparer le `margin-top` pour qu'il soit à 3px uniquement pour la home page. La raison est qu'avec 5px présentement le bas du bouton bleu arrive "flush" avec le bas du bouton de menu, comme ceci : 
Screen Shot 2022-08-29 at 3 57 17 PM
Ce serait donc bien de garder le 5px pour toutes les pages SAUF la home page, en ayant un nouveau bloc de style :
```
.home #wb-so {
	.btn {
		margin-top: 3px;
	}
}
```

File: templates/home/layouts/home.html [hunk 2]
```diff
@@ -37,15 +37,15 @@
...
+					<div class="col-md-8">
 				{%- else -%}
 					<div class="col-sm-12">
 				{%- endif -%}
 				{% include gcweb-menu/sitemenu.html %}
 			{%- endunless -%}
 		</div>
 		{%- if page.auth -%}
-			<div class="col-sm-3">
+			<div class="col-xs-5 col-xs-offset-7 col-md-offset-0 col-md-4">
```
Comment (GormFrank) [inline templates/home/layouts/home.html:48]: These changes will be included as part of #1977 as well, so this is just a note for ourselves to take this into considerations. Documentation for this change on the home page however is in the other PR while this current PR is more pressing. One other thing to consider though is the fact that we simply forgot this template when we first implemented the Contextual sign in button.

CC: @Garneauma @Ricokola

**Review by GormFrank — CHANGES_REQUESTED** [review]

**Review by duboisp — CHANGES_REQUESTED** [review]
move the working example "contextual-signin-home" into the templates/home
but keep the link to it from the "sites/authentification/index.json-ld" file

Add a deprecated example of the current version of the home into "templates/home/deprecated". 
Move the deprecated home v1 (currently here: /sites/header/deprecated/home-v1.html ) into the "templates/home/deprecated" folder.

Add link to the home page deprecated version from the home page documentation (templates/home/api-en.html && api-fr.html)

**Review by Ricokola — DISMISSED** [review]
- This PR have few conflicts with PR# 1977 and PR# 2001 to take in consideration.
- The working example for deprecated home page v2 including the authentication section is not appearing correctly. The button is on a second line and touching the banner:  templates/home/deprecated/home-v2-en.html but the French version is ok. Also, the language toggle doesn't work for both version.
-

File: sites/authentication/_base.scss [hunk 3]
```diff
@@ -7,6 +7,10 @@
 	.btn {
 		border-radius: 0px;
 		margin-top: 5px;
+
+		.home & {
```
Comment (duboisp) [inline sites/authentication/_base.scss:11]: This belong to the home page template not to the authentication generic pattern.

File [home.css](https://github.com/wet-boew/GCWeb/blob/master/templates/home/_base.scss#L12)

.home #wb-so .btn {
   margin-top: 3px;
}

**Review by duboisp — CHANGES_REQUESTED** [review]

File: sites/authentication/_base.scss (hunk 3 above)
Comment (Garneauma) [reply sites/authentication/_base.scss:11]: Le SASS actuel est compilé de la manière suivante en CSS: 

`#wb-so .btn {
  border-radius: 0px;
  margin-top: 5px;
}
.home #wb-so .btn {
  margin-top: 3px;
}
#wb-so a.btn:hover {
  background-color: #444;
}`

Il permet donc d'effectuer la même commande, mais plus simplement.

File: templates/home/contextual-signin-home-en.html [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+	"title": "Home - Canada.ca",
+	"language": "en",
+	"altLangPage": "contextual-signin-home-fr.html",
+	"dateModified": "2022-08-10",
+	"layout": "home",
+	"noReportProblem": true,
+	"auth": {
+		"type": "contextual",
+		"link": "https://www.canada.ca/en/revenue-agency/services/e-services/e-services-individuals/account-individuals.html"
+	}
```
Comment (duboisp) [inline templates/home/contextual-signin-home-en.html:12]: Move this config on the main home page example. 

Unless if we are going to apply that new pattern in a far future, please update directly the main home page working example.

**Review by duboisp — CHANGES_REQUESTED** [review]
Once the working example is merged/updated. good to go for me.

File: templates/home/deprecated/home-v2-en.html [hunk 5]
```diff
@@ -0,0 +1,336 @@
...
+							<li role="presentation"><a role="menuitem" href="https://www.canada.ca/en/services/science.html">Science and innovation</a></li>
+						</ul>
+					</nav>
+				</div>
+				<div class="col-sm-9">
+					<section id="wb-so">
+						<h2 class="wb-inv">Sign in</h2>
+						<a class="btn btn-primary" href="https://www.canada.ca/en/revenue-agency/services/e-services/e-services-individuals/account-individuals.html">Sign in</a>
+					</section>
+				</div>
```
Comment (duboisp) [inline templates/home/deprecated/home-v2-en.html:100]: There is no "Sign-in" for the version 2 of the home page

File: templates/home/deprecated/home-v2-en.html [hunk 6]
```diff
@@ -0,0 +1,336 @@
...
+		</ul>
+	</nav>
+	<header>
+		<div id="wb-bnr" class="container">
+			<div class="row">
+				<section id="wb-lng" class="col-xs-3 col-sm-12 pull-right text-right">
+					<h2 class="wb-inv">Language selection</h2>
+					<ul class="list-inline mrgn-bttm-0">
+						<li>
+							<a lang="fr" hreflang="fr" href="contextual-signin-home-fr.html">
```
Comment (duboisp) [inline templates/home/deprecated/home-v2-en.html:37]:
```suggestion
							<a lang="fr" hreflang="fr" href="home-v2-fr.html">
```

File: templates/home/deprecated/home-v2-fr.html [hunk 7]
```diff
@@ -0,0 +1,332 @@
...
+							</button>
+						</div>
+					</form>
+				</section>
+			</div>
+		</div>
+		<hr>
+		<div class="container">
+			<div class="row">
+				<div class="col-sm-9">
```
Comment (duboisp) [inline templates/home/deprecated/home-v2-fr.html:72]: This is the CSS grid configuration for the version 3 not for the version 2 which should say `<div class="col-sm-12">`

File: templates/home/deprecated/home-v2-fr.html [hunk 8]
```diff
@@ -0,0 +1,332 @@
...
+							<li role="presentation"><a role="menuitem" href="https://www.canada.ca/fr/services/science.html">Science et innovation</a></li>
+						</ul>
+					</nav>
+				</div>
+				<div class="col-sm-3">
+					<section id="wb-so">
+						<h2 class="wb-inv">Connexion</h2>
+						<a class="btn btn-primary" href="https://www.canada.ca/fr/agence-revenu/services/services-electroniques/services-electroniques-particuliers/dossier-particuliers.html">Connexion</a>
+					</section>
+				</div>
```
Comment (duboisp) [inline templates/home/deprecated/home-v2-fr.html:100]: There is no "sign in" in version 2

File: templates/home/deprecated/home-v2-fr.html [hunk 9]
```diff
@@ -0,0 +1,332 @@
...
+		</ul>
+	</nav>
+	<header>
+		<div id="wb-bnr" class="container">
+			<div class="row">
+				<section id="wb-lng" class="col-xs-3 col-sm-12 pull-right text-right">
+					<h2 class="wb-inv">Sélection de la langue</h2>
+					<ul class="list-inline mrgn-bttm-0">
+						<li>
+							<a lang="en" hreflang="en" href="contextual-signin-home-en.html">
```
Comment (duboisp) [inline templates/home/deprecated/home-v2-fr.html:37]:
```suggestion
							<a lang="en" hreflang="en" href="home-v2-en.html">
```

**Review by duboisp — CHANGES_REQUESTED** [review]
I confirm the change in the home layout are the same as the sites/header/includes/header.html

Regarding the few small requested change, @Garneauma  let us know if it is not possible to have that completed first thing tomorrow.

There is also a merge conflict to resolve.

After that, this PR looks good to merged.

\cc @GormFrank

**Review by duboisp — APPROVED** [review]

**Pull request merged**